### DISCUSSION TITLE
Compute all Hidden Activations

### DISCUSSION THREAD

**Discussion by Whadup** [microsoft/onnxruntime#5703] [Other Q&A]
Hi there,
how can I use onnxruntime to compute not only the output nodes, but get the values of all nodes of the computation graph?
I can't find anything on the topic, hope you can help.
Best
Lukas

**Comment by pranavsharma** [comment]
What's your use case? Why do you want to do this? If this is just for debugging see https://github.com/microsoft/onnxruntime/blob/master/BUILD.md#DebugNodeInputsOutputs. If this is not for debugging, add all the required nodes as outputs of the graph.

**Reply by Whadup** [reply]
I added all the nodes to the outputs using the `onnx`library. 
https://github.com/microsoft/onnxruntime/issues/2119
thanks
